pkg/aggregator/check_sampler.go

```diff
@@ -107,35 +105,16 @@ func (cs *CheckSampler) addBucket(bucket *metrics.HistogramBucket) {
 		// noop
 		return
 	}
-
-	// simple linear interpolation, TODO: optimize
-	var linearIncr float64
-	var incrCount int
-	var countPerIncr uint
-	if bucket.Value > cs.interpolationGranularity {
-		linearIncr = bucketRange / float64(cs.interpolationGranularity)
-		countPerIncr = uint(bucket.Value / cs.interpolationGranularity)
-		incrCount = cs.interpolationGranularity
-	} else {
-		linearIncr = bucketRange / float64(bucket.Value)
-		countPerIncr = 1
-		incrCount = bucket.Value
-	}
 	if math.IsInf(bucket.UpperBound, 1) {
-		// We simulate the behavior of promQL for the infinity bucket:
-		// "if the quantile falls into the highest bucket, the upper bound of the 2nd highest bucket is returned"
-		incrCount = 1
-		countPerIncr = uint(bucket.Value)
+		cs.sketchMap.insertInterp(int64(bucket.Timestamp), contextKey, bucket.LowerBound, bucket.LowerBound, uint(bucket.Value))
```

Comment:
Could we add a comment on why we're doing this?
I had to double check what was the difference between the line bellow until I noticed the double `LowerBound` 😅